components/Choose/index.tsx

```diff
@@ -10,12 +10,21 @@ import {
   ErrorBox,
   ShadowBox,
   SmartLink,
-  LoadingScreen
+  LoadingScreen,
+  LoadingIcon
 } from '@glif/react-components'
+import styled from 'styled-components'
 
 import { useMsig } from '../../MsigProvider'
 import { PAGE, QPARAM } from '../../constants'
 
+const LoadingBox = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  gap: var(--space-l);
+`
```

Comment:
I see that we already include a `LoadingScreen` for this component. Event though the conditions of these are slightly different, I think it'd be nicer to use one type of loading indication and not two.

If we stick with `LoadingBox`, the name is slightly misleading to me because all of our current Box components (WarningBox, PrimaryBox, etc) look like boxes and this is just a div wrapper.

I'd also change update the condition of showing the loading indicator slightly, because this is now what happens for me:
- I enter the safe ID and press Submit
- I see the loading indicator
- I see the Safe ID input screen flash by again
- Then I go to the safe home

The last flash of the Safe input should be prevented